Plugins update manager: Edit schedule

bogiii opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->
Relates https://github.com/Automattic/dotcom-forge/issues/5544
Closes https://github.com/Automattic/dotcom-forge/issues/5645

## Proposed Changes

* Implemented schedule update logic
* Introduced a new `edit` route in format: `/plugins/scheduled-updates/:action/:site_slug?/:schedule_id`

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Go to `plugins/scheduled-updates/{ATOMIC SITE}`
* Try create/update/delete of schedules and check if it works

**NOTE:** https://github.com/Automattic/wp-calypso/pull/88198 will cover the submit button busy state and UX improvements of the whole CRUD process, such as waiting for the server to delete an item from the list, etc.

![Screen Capture on 2024-03-05 at 13-46-45](https://github.com/Automattic/wp-calypso/assets/1241413/6b740994-919a-44a1-917c-af72f955f2ab)


## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

bogiii committed:
Bootstrap edit route

bogiii committed:
Set callback method param to be optional

bogiii committed:
Consume const from the config file

bogiii committed:
Get schedule object for the edit context

bogiii committed:
Create edit schedule mutation hook

bogiii committed:
Implement schedule edit functionality

bogiii committed:
Rename callback param

vishnugopal commented on client/blocks/plugins-update-manager/index.tsx:
```diff
@@ -39,14 +42,22 @@ export const PluginsUpdateManager = ( props: Props ) => {
+					edit: (
+						<ScheduleEdit siteSlug={ siteSlug } scheduleId={ scheduleId } onNavBack={ onNavBack } />
+					),
+				}[ context ]
```
Neat, I think this is the first time I'm seeing this kind of usage in JSX. Cleaner than the conditionals!

bogiii replied:
Yeah, I like it as well. Much better than the conditions.

vishnugopal commented on client/blocks/plugins-update-manager/schedule-edit.tsx:
```diff
@@ -0,0 +1,59 @@
+			</CardBody>
+			<CardFooter>
+				<Button form="schedule" type="submit" variant="primary">
+					Edit
```
I think by convention this should say Save instead of Edit.

bogiii replied:
Sure, I'll update it.

vishnugopal reviewed (changes requested):
Unrelated to this PR, but I noticed that we may need to process entities correctly in the plugin list: 
![CleanShot 2024-03-06 at 09 59 52](https://github.com/Automattic/wp-calypso/assets/533/b2baedff-dfe2-43d5-9407-701a74186a28)

I think we should have an Edit action in the dropdown as well especially as the entire row does not seem clickable:
![CleanShot 2024-03-06 at 10 02 07](https://github.com/Automattic/wp-calypso/assets/533/c16c703d-bf32-4344-9acd-2a940da16ca4)

bogiii committed:
Resolve conflict issue

bogiii committed:
Update submit button text from 'Update' to 'Save'

bogiii committed:
Add 'Edit' dropdown action

bogiii commented:
> I think we should have an Edit action in the dropdown as well especially as the entire row does not seem clickable:

Agree. I updated the list of actions and added 'Edit'.


> Unrelated to this PR, but I noticed that we may need to process entities correctly in the plugin list:

I'll fix it in the next PR.

bogiii committed:
Fix typing error

vishnugopal reviewed (approved):
Approved. One unit test is failing below, but I think it's unrelated.

bogiii commented:
> One unit test is failing below, but I think it's unrelated.

It's fixed; it was a typing error.

Thanks, @vishnugopal

bogiii committed:
Fix type error
